### DISCUSSION TITLE
Consolidate multiple backups

### DISCUSSION THREAD

**Discussion by datadecoded** [tmo1/sms-ie#226] [Q&A] [answered]
Hi @tmo1  - thanks again for all of your work on this app, it's been super useful! Do you have any suggestions for how to consolidate multiple backups outside of the app? For the previous version of the JSON, I could just use python to combine, but not sure about the updated format.

**Comment by tmo1** [comment, marked as answer]
> thanks again for all of your work on this app, it's been super useful!

You're very welcome - I'm glad the app has been useful!

> Do you have any suggestions for how to consolidate multiple backups outside of the app? For the previous version of the JSON, I could just use python to combine, but not sure about the updated format.

Whatever you did before should work for the new format as well, with appropriate modifications. There are two main differences between the formats:

1. The old format used regular JSON, while the new one uses NDJSON. "NDJSON" is simpler than it sounds - it's just individual lines of JSON, one for each message, separated by newlines (with no newline separators between tags and elements of the JSON for each message).
2. The old format embedded binary MMS parts (base64 encoded) into the JSON; the new format keeps them as separate files.

The details of the new format are documented [here](https://github.com/tmo1/sms-ie/#ndjson-structure). I'm happy to provide further information as desired.
